File: inject-java/src/test/groovy/io/micronaut/aop/introduction/InterfaceIntroductionClass.java
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.aop.introduction;

import io.micronaut.aop.simple.Mutating;

import javax.inject.Singleton;

/**
 * @author Graeme Rocher
 * @since 1.0
 */
@Stub
@Mutating("name")
@Singleton
public interface InterfaceIntroductionClass<A> extends SuperInterface<A> {


    String test(String name);

    String test(String name, int age);

}

File: test-inject-kotlin2-ksp2/src/test/kotlin/io/micronaut/kotlin/processing/aop/introduction/InterfaceIntroductionClass.kt
package io.micronaut.kotlin.processing.aop.introduction

import io.micronaut.kotlin.processing.aop.simple.Mutating
import jakarta.inject.Singleton

@Stub
@Mutating("name")
@Singleton
interface InterfaceIntroductionClass<A> : SuperInterface<A> {

    fun test(name: String): String
    fun test(name: String, age: Int): String
}
